inst/private/doctest_collect.m

```diff
@@ -234,120 +234,141 @@
   docstring = '';
   error = '';
 
-  % strip @group, and escape sequences
-  str = regexprep(str, '^\s*@group\n', '\n', 'lineanchors');
-  str = regexprep(str, '@end group\n', '');
-  str = strrep(str, '@{', '{');
-  str = strrep(str, '@}', '}');
-  str = strrep(str, '@@', '@');
-
-  % special comments "@c doctest: cmd" are translated
-  re = [ '@c(?:omment)?'    ...  % @c or @comment, ?: means no token
-         '\s*(?:#|%|\s)\s*' ...  % at least one space or one of #,%
-         '(doctest:\s*.*\n)' ];  % want the doctest token
-  str = regexprep(str, re, '% $1', 'dotexceptnewline');
-
-  % texinfo comments: drop remainder of line
-  str = regexprep(str, '@c(omment)?\s+.*\n', '\n', 'dotexceptnewline');
-
   % no example blocks? not an error, but nothing to do
   if (isempty(strfind(str, '@example')))
     % error = 'no @example blocks';
     return
   end
 
-  % leave the @example lines in, may need them later
-  T = regexp(str, '(@example.*?@end example)', 'tokens');
-  if (isempty(T))
-    error = 'malformed @example blocks';
+  % Mark the occurrence of “@example” and “@end example” to be able to find
+  % example blocks after conversion from texi to plain text.  Also consider
+  % indentation, so we can later correctly unindent the example's content.
+  % There seems to be a bug with substitute replacement in the first line and
+  % the second pattern could fail if the file doesn't end with a newline.  Thus
+  % we prepend and append a newline.
+  str = regexprep (cstrcat (char (10), str, char (10)), ...
+                   '^([ \t]*)(@example)(.*)(\r?\n)', ...
```

Comment:
No, this would not work, because \s matches newline characters and then $1 would introduce additional lines.

I currently don't know how to make the newline handling any better. The TexInfo code requires many command within dedicated lines and I have to be careful to not accidentally split or join the lines.